sql-cli/src/sql_cli/dag_generator.py

```diff
@@ -0,0 +1,46 @@
+from __future__ import annotations
+
+from networkx import DiGraph, depth_first_search, find_cycle, is_directed_acyclic_graph
+from sql_directory_parser import SqlFile
+
+
+class SqlFilesDAG:
```

Comment:
Can this same class be used as the backend for an eventual render function as well?

Maybe we can factor out certain elements of this and put it into the python-sdk module for that function?

Reply:
Interesting idea. 🤔

Reply:
I am constantly refactoring the code. Let's have a look again when it is a bit more mature.